### PR TITLE
Fix Bluebird Promise.allSettled non-standard polyfill

### PR DESCRIPTION
**Pull request by TanukiSharp**
@iartem (CC @ar2rsawseen)

Hi Artem, I would like to make a suggestion for a different way to handle Bluebird non-standard `Promise.allSettled`.

### CODE REVIEW

File: plugins/pluginManager.js [hunk 1]
```diff
@@ -421,6 +421,19 @@ var pluginManager = function pluginManager() {
...
+        return bluebirdResults.map(bluebirdResult => {
+            const isFulfilled = bluebirdResult.isFulfilled();
+
+            const status = isFulfilled ? 'fulfilled' : 'rejected';
+            const value = isFulfilled ? bluebirdResult.value() : undefined;
+            const reason = isFulfilled ? undefined : bluebirdResult.reason();
+
+            return { status, value, reason };
+        });
+    }
```
Comment (ar2rsawseen) [inline plugins/pluginManager.js:435]: Codacy found an issue: [Missing semicolon.](https://app.codacy.com/manual/ar2rsawseen/countly-server/pullRequest?prid=5769062)

**Comment by ar2rsawseen** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 3
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ar2rsawseen/countly-server/pullRequest?prid=5769062&bid=18880282)

File: plugins/pluginManager.js [hunk 2]
```diff
@@ -421,6 +421,19 @@ var pluginManager = function pluginManager() {
         events[event].push(callback);
     };
 
+    // TODO: Remove this function and all it calls when moving to Node 12.
+    var promiseAllSettledBluebirdToStandard = function(bluebirdResults) {
+        return bluebirdResults.map(bluebirdResult => {
```
Comment (ar2rsawseen) [inline plugins/pluginManager.js:426]: Codacy found an issue: [Expected parentheses around arrow function argument.](https://app.codacy.com/manual/ar2rsawseen/countly-server/pullRequest?prid=5769062)

**Comment by ar2rsawseen** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ar2rsawseen/countly-server/pullRequest?prid=5769062&bid=18880282)

**Comment by iartem** [comment]
I think that won't work as follow-on code expects only results ignoring all the errors, without status.

**Comment by mt-sebastien-robert** [comment]
Yeah and that's a problem, there are some scenarios where errors must be dealt with, so I think Countly should not swallow them, but let the caller deal with them.

**Comment by iartem** [comment]
Sorry but I cannot merge that as it would break apps updates, for example. I'm sure there are a plenty of other places like that.

**Comment by mt-sebastien-robert** [comment]
This changes already broke the previous behavior: https://github.com/Countly/countly-server/commit/2e2bfee9a652bc6d98db8c6a2c6469fe79268c82#diff-b94338ae957cdd5c7365874473305f41

**Pull request merged**